### ISSUE TITLE
Enhancement: Configurable slider highlighting

### ISSUE THREAD

**Issue by bpatterson88** [Esri/calcite-design-system#1631]
### Description

Allow slider to highlight different sections of the rail, ex. left of the handle _or_ right of the handle. 

### Acceptance Criteria 

- Single value: rail can highlight right, left, or none

### Relevant Info 

Dashboards currently uses sliders to filter visualizations, and depending on the operator we highlight different parts of the rail. This request will give Calcite parity with the current Dashboards implementation of slider.

#### Which Component

Slider

#### Example Use Case

- Select all values greater than _N_ (ex. to the right of the handle)
- Select _N_ (ex. no highlighting)

**Comment by paulcpederson** [comment]
@bpatterson88  would this need to apply to range as well or just single value?

**Comment by bpatterson88** [comment]
@paulcpederson Nope, just the single value.

Also, updated the request to include a 'none' option for highlighting, where the slider is used to select a single value.

**Comment by bpatterson88** [comment]
@benelan Dashboards uses sliders for less than, greater than, and equals queries. Different parts of the slider rail are highlighted depending on the operator. The screenshots below show our current implementation. 

![Screen Shot 2021-11-29 at 4 30 29 PM](https://user-images.githubusercontent.com/15875886/143963885-9c73b512-229b-4e6e-b47e-568d50357f1e.png)
![Screen Shot 2021-11-29 at 4 30 38 PM](https://user-images.githubusercontent.com/15875886/143963889-92113fd5-3197-46a3-919e-290f27d3d722.png)
![Screen Shot 2021-11-29 at 4 30 47 PM](https://user-images.githubusercontent.com/15875886/143963896-69dd8beb-b076-4acc-938b-dbb13169fe28.png)

**Comment by ellenupp** [comment]
Hi @benelan, just wanted to put this issue on your radar again and see where it's at. We would love to start using calcite-slider in dashboards but this is a blocker for us.

**Comment by benelan** [comment]
@bpatterson88 thanks for providing the use case, and thanks for bumping the issue @ellenupp. We are planning to get this in before v1 and will have a better timeline after our next triage meeting, to determine which sprint has cycles.

**Comment by richiecarmichael** [comment]
This is needed for the new JSAPI **Range Slider** (see [#46032](https://devtopia.esri.com/WebGIS/arcgis-js-api/issues/46032)).

**Comment by ubatsukh** [comment]
@driskull  as we discussed yesterday this is a high priority issue for us! MV needs Range Slider for multidimensional image services and the API implementation cannot proceed until the calcite slider has support for this.

**Comment by ellenupp** [comment]
Hello, any updates on this issue? We still need this for Dashboards.

**Comment by ashetland** [comment]
@jcfranco Thoughts on this? Some possible prop names:

`highlight = none | greater-than | less-than`
or
`active-track = none | greater-than | less-than`

`active-track` feels the most correct, but not sure if we already have a convention for this one.

**Comment by jcfranco** [comment]
I like `highlight` since it's both expressive and concise. `range-highlight` is too verbose, right? 😛

**Comment by SkyeSeitz** [comment]
It sounds like these would all still fall under the non-range slider- correct? if so, I think the `range-highlight` prop might confuse the distinction.
<img width="876" alt="image" src="https://github.com/Esri/calcite-components/assets/29716057/86b7a671-b04f-4fc6-b767-b8c3b00f1ce6">

**Comment by SkyeSeitz** [comment]
One other note to confirm- this would mean all current single value sliders would change in appearance from the left design with a highlighted track to the right design with no highlight? 
<img width="1125" alt="image" src="https://github.com/Esri/calcite-components/assets/29716057/93507d12-9b47-4190-b29d-82272df64e23">

**Comment by ashetland** [comment]
Based on our discussion, yes it would seem that this would make the default `highlight=none` which might be considered a breaking change for folks.

And yes, just to confirm, this would only apply to the single slider, not range, per the comments above.

**Comment by jcfranco** [comment]
> It sounds like these would all still fall under the single value slider- correct? if so, I think the range-highlight prop name might confuse the distinction.

Good point. I wasn't considering the range mode. How would `highlight` options interact with this mode?

> this would mean all current single value sliders would change in appearance from the left design with a highlighted track to the right design with no highlight?
> Based on our discussion, yes it would seem that this would make the default highlight=none which might be considered a breaking change for folks.

Is this because the proposed default would be `"none"` or is there another reason? Wondering if we could make `less-than` the default to minimize visual differences.

**Comment by SkyeSeitz** [comment]
Stepping back and evaluating the greater than/less than use case, it ultimately functions more as a range with a fixed start or end because it is outputting everything between two values. Adding something along the lines of a `fixed-start/end` prop as a range enhancement would avoid avoid confusion and unwanted overrides to current default Sliders - changing them to a less than function or changing their appearance to no highlight. 

I have mapped out that story below (design details for fixed-start/end could use work), but would be good to get confirmation from Dashboards on their use case first. 

<img width="1268" alt="image" src="https://github.com/Esri/calcite-components/assets/29716057/37cd4fbf-dac8-4fe6-afac-c9b434406fba">

**Comment by ellenupp** [comment]
Thanks for the quick work on this!

We also have a use for range sliders but we only ever want to highlight between the two handles in that case. So this enhancement would just be for the single value case, when we want to highlight either the left/right/neither side of the track depending on the operation the value is used in. 

Something like `highlight = none | greater-than | less-than` where this only has an effect when there's a single value would do the trick. 

> changing them to a less than function

I'm just envisioning this to be a visual representation, no change to the actual value or function of the slider.

**Comment by SkyeSeitz** [comment]
Had a good convo with Ellen and Aaron this morning and got clarification on a single value Slider needing the highlight to sit on either side of the defined value. Below is the updated proposal for a new `highlight` prop with two values `"none | mirrored"`. The default Slider behavior can remain untouched with no additional props. 

<img width="1006" alt="image" src="https://github.com/Esri/calcite-components/assets/29716057/b1bccea1-6ca2-4576-b1ea-55d0178c50cc">

Dashboards is using the mirrored highlight in combination with the default Slider in a greater-than/less-than context, but the highlight prop provided would simply be a visual change to remove the highlight or flip it.

cc @jcfranco for dev input

**Comment by geospatialem** [comment]
Effort could be joined with https://github.com/Esri/calcite-components/issues/2991

**Comment by DitwanP** [comment]
🍬 Verified in `2.8.0-next.7`

**Comment by ellenupp** [comment]
Awesome, thank you!

But I noticed when ticks are enabled they aren't respecting the highlight-placement
![image](https://github.com/Esri/calcite-design-system/assets/14304581/6820ae87-4d90-45ad-a8da-a428a628d4a2)

**Comment by DitwanP** [comment]
Reopening to fix the issue with ticks mentioned above. [Codepen](https://codepen.io/Ditwan-Price/pen/JjVzXRX?editors=1000)

**Comment by jcfranco** [comment]
@ellenupp Thanks for catching this. ✨🎣✨

I also noted that the prop should've been named `fill-placement` vs `highlight-placement`. I'll submit a follow-up PR for this as well.

**Comment by DitwanP** [comment]
Verified on `2.8.0-next.12`
 
![image](https://github.com/Esri/calcite-design-system/assets/60022782/bc6bcd38-0ea9-4778-a2bb-aa5c90f0d38e)

**Comment by jcfranco** [comment]
Installed! That should be the last of it. 🤞 #famouslastwords

Thanks everyone and sorry for the hassle!

**Comment by DitwanP** [comment]
😬🤞 Verified on `2.8.0-next.14`
